scripts/proteinInteractionMINT/parse_mint_test.py

```diff
@@ -0,0 +1,101 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test for parse_mint.py.
+Run "python3 parse_mint_test.py"
+"""
+
+import unittest
+import parse_mint
+
+CONST_INPUT = ('uniprotkb:Q04439\tuniprotkb:Q12446\tintact:EBI-11687|uniprotkb:D6VZT2\tintact:'
+               'EBI-10022|uniprotkb:D6W2N7\tpsi-mi:myo5_yeast(display_long)|uniprotkb:'
+               'Class I unconventional myosin MYO5(gene name synonym)|uniprotkb:Type '
+               'I myosin MYO5(gene name synonym)|uniprotkb:Actin-dependent myosin-I '
+               'MYO5(gene name synonym)|uniprotkb:MYO5(gene name)|psi-mi:MYO5(display_short)'
+               '|uniprotkb:YMR109W(locus name)|uniprotkb:YM9718.08(orf name)\tpsi-mi:las17_yeast'
+               '(display_long)|uniprotkb:YOR181W(locus name)|uniprotkb:LAS17(gene name)|psi-mi:'
+               'LAS17(display_short)\tpsi-mi:"MI:0084"(phage display)\tTong et al. (2002)\tmint:'
+               'MINT-5214930|pubmed:11743162\ttaxid:559292(yeast)|taxid:559292(Saccharomyces '
+               'cerevisiae)\ttaxid:559292(yeast)|taxid:559292(Saccharomyces cerevisiae)\t'
+               'psi-mi:"MI:0915"(physical association)\tpsi-mi:"MI:0471"(MINT)\tintact:'
+               'EBI-606428\tauthor score:Above 1.|intact-miscore:0.81\nuniprotkb:P38764'
+               '\tuniprotkb:P40016\tintact:EBI-15913|uniprotkb:D3DKX4\tintact:EBI-15927'
+               '|uniprotkb:D3DLS0\tpsi-mi:rpn1_yeast(display_long)|uniprotkb:Proteasome '
+               'non-ATPase subunit 1(gene name synonym)|uniprotkb:HMG-CoA reductase '
+               'degradation protein 2(gene name synonym)|uniprotkb:YHR027C(locus name)'
+               '|uniprotkb:RPN1(gene name)|psi-mi:RPN1(display_short)|uniprotkb:HRD2'
+               '(gene name synonym)|uniprotkb:NAS1(gene name synonym)|uniprotkb:RPD1'
+               '(gene name synonym)\tpsi-mi:rpn3_yeast(display_long)|uniprotkb:YER021W'
+               '(locus name)|uniprotkb:RPN3(gene name)|psi-mi:RPN3(display_short)|'
+               'uniprotkb:SUN2(gene name synonym)\tpsi-mi:"MI:0676"(tandem affinity purification)'
+               '\tKrogan et al. (2006)\tpubmed:16554755|imex:IM-15332|mint:MINT-5218454\ttaxid:'
+               '559292(yeast)|taxid:559292(Saccharomyces cerevisiae)\ttaxid:559292(yeast)|taxid:'
+               '559292(Saccharomyces cerevisiae)\tpsi-mi:"MI:0915"(physical association)\tpsi-mi:'
+               '"MI:0471"(MINT)\tintact:EBI-6941860|mint:MINT-1984371|imex:IM-15332-8532\t'
+               'intact-miscore:0.76')
+
```

Comment:
nit: Consider making this more readable as:

const input = ("""
uniprotkb:Q04439    uniprotkb:...
""")

And see if just copy-pasting the tabs as-is works.

Reply:
It seems the tab '\t' is causing the problem. Example: CONST_INPUT = '''uniprotkb:Q04439	uniprotkb:Q12446	intact:EBI-11687|uniprotkb:D6VZT2	intact:EBI-10022|uniprotkb:D6W2N7	psi-mi:myo5_yeast(display_long)|uniprotkb:Class I unconventional myosin MYO5(gene name synonym)|uniprotkb:Type I myosin MYO5(gene name synonym)|uniprotkb:Actin-dependent myosin...'''